Add MM upstreaming guide to GitHub (#5057)
terraform-providers/terraform-provider-google: .github/CONTRIBUTING.md

your `.bashrc` or `.bash_profile`.

```bash
function tpgpatch1 {
  pr_username=$(echo $1 | cut -d ':' -f1)
  feature_branch=$(echo $1 | cut -d ':' -f2)
  git remote add $pr_username git@github.com:$pr_username/${PWD##*/}
  git fetch $pr_username
  git checkout $pr_username/$feature_branch
  git format-patch $(git merge-base HEAD master)
}
 
function tpgpatch2 {
  for patch in $GOPATH/src/github.com/terraform-providers/terraform-provider-google*/*.patch; do
    echo "checking ${patch}"
        if git apply --stat $patch | grep "google/"; then
                git am -3 -i $patch -p2 --directory=third_party/terraform/resources/ --include="*.go"
        fi
        if git apply --stat $patch | grep "google-beta/"; then
                git am -3 -i $patch -p2 --directory=third_party/terraform/resources/ --include="*.go"
        fi
        if git apply --stat $patch | grep "markdown"; then
                git am -3 -i $patch --directory=third_party/terraform/ --include="*website/*"
        fi
  done
}
```

With those functions defined:

1. Check out both the provider and MM repo to `master`, committing/stashing any local changes

> 1. Run `git checkout -b {{branch}}` to create a branch for your upstreaming PR

Reviewer: ```suggestion
1. In the MM repo, run `git checkout -b {{branch}}` to create a branch for your upstreaming PR
```

1. Click the clipboard button next to the `author:branch` indicator on the PR to copy it.
1. Run `tpgpatch1 author:branch` from the provider repo

> 1. Run `tpgpatch2 author:branch` from the MM repo

Reviewer: ```suggestion
1. Run `tpgpatch2` from the MM repo
```

1. Remove the patch files from the provider repo

At this point, you should be checked out to a branch with the changes to handwritten files included in the MM repo. For

> generated and most compiled filed, you'll need to perform the upstreaming manually. Following getting your local branch

Reviewer: ```suggestion
generated files and most compiled files, you'll need to perform the upstreaming manually. After getting your local branch
```
